### PR TITLE
Enhancement--Script deployment support. Issue 926

### PR DESCRIPTION
**Pull request by kwekmaster**
Addresses https://github.com/WolvenKit/WolvenKit/issues/926

- Specification discussion was done earlier thru discord and can be ref'd if needed.
- Supports all present and future scripting elements--assuming they are deployed under the game directory.  
- The sample testbed I used contain CET, redscript, and red4ext dlls for deployment.

**Implemented**

_Assumptions for the preliminary approach: the user is responsible for structuring their scripts to match the target deployment directory structure._

- The ability to deploy scripting elements user puts under `resource` into `packed`. Please see the testbed below for a sample structure.
- Simple as-is copying, skipping the special `yaml` and `xl` files.

**Notes**
Hot deployment seems to follow a separate path and is not touched by this PR.

**Testing Done**
- Using everything in the test bed below
- With only yaml/xl files
- With blank `resources`
- Both as regular and REDmod 

Testing meets the requirements.

**Test Bed**
Tests are done the following `resources` structure. `.xl` and `.yaml` files are added thru clicking `(+)New File`
image

### CODE REVIEW

File: WolvenKit.App/Controllers/RED4Controller.cs [hunk 1]
```diff
@@ -463,6 +475,33 @@ private bool PackArchives(Cp77Project cp77Proj, LaunchProfile options)
 
             return true;
         }
+
+
+        private static bool PackScriptFiles(Cp77Project cp77Proj)
+        {
+            //preliminary approach -- everything under resources, except for *.yaml, *.xl, is a 'script' element
```
Comment (rfuzzo) [inline WolvenKit.App/Controllers/RED4Controller.cs:482]: hmm, this doesn't seem right? Scripts should only be ".reds" files... I know it's preliminary, but I think this is confusing. You could change the name of the function to "PackResources" I suppose.

**Review by rfuzzo — CHANGES_REQUESTED** [review]

File: WolvenKit.App/Controllers/RED4Controller.cs (hunk 1 above)
Comment (kwekmaster) [reply WolvenKit.App/Controllers/RED4Controller.cs:482]: Although I have concerns about Scripts == `*.reds` exclusively, I do agree that this preliminary approach packs resources. I will change the method to "PackResources" so we have a path for users (like me) to deploy their scripts :)

The change will be made.

Comment (kwekmaster) [reply WolvenKit.App/Controllers/RED4Controller.cs:482]: Changes completed!  Please review.

**Review by rfuzzo — APPROVED** [review]
thank you!

**Pull request merged**